exampleconnectionservice/src/main/java/com/twilio/voice/exampleconnectionservice/VoiceActivity.java

```diff
@@ -0,0 +1,370 @@
+
+        //noinspection SimplifiableIfStatement
+        if (id == R.id.action_settings) {
+            return true;
+        }
+
+        return super.onOptionsItemSelected(item);
+    }
+
+    /*
+    * Setup a phone Account
+    */
+    void setupPhoneAccount() {
+        String appName = this.getString(R.string.connection_service_name);
+        handle = new PhoneAccountHandle(new ComponentName(this.getApplicationContext(), VoiceConnectionService.class), appName);
+        telecomManager = (TelecomManager) this.getApplicationContext().getSystemService(this.getApplicationContext().TELECOM_SERVICE);
+        phoneAccount = new PhoneAccount.Builder(handle, appName)
+                .setCapabilities(PhoneAccount.CAPABILITY_CALL_PROVIDER)
+                .build();
+        telecomManager.registerPhoneAccount(phoneAccount);
+    }
+
+    // Request required permissions
+    public static boolean hasPermissions(Context context, String... permissions) {
+        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && context != null && permissions != null) {
+            for (String permission : permissions) {
+                if (ActivityCompat.checkSelfPermission(context, permission) != PackageManager.PERMISSION_GRANTED) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        if (requestCode == MIC_PERMISSION_REQUEST_CODE && permissions.length > 0) {
+            if (grantResults[0] != PackageManager.PERMISSION_GRANTED) {
+                Snackbar.make(coordinatorLayout,
+                        "Microphone permissions needed. Please allow in your application settings.",
+                        SNACKBAR_DURATION).show();
+            }
+        } else if (requestCode == CALL_PHONE_CODE && permissions.length > 0) {
+            if (grantResults[0] != PackageManager.PERMISSION_GRANTED) {
+                Snackbar.make(coordinatorLayout,
+                        "CALL_PHONE permissions needed. Please allow in your application settings.",
+                        SNACKBAR_DURATION).show();
+            }
+        }
+
+        if (hasPermissions(this, PERMISSIONS)) {
+            retrieveAccessToken();
+        }
+    }
+
+    /*
+     * Setup the broadcast receiver to be notified of FCM Token updates
+     * or incoming call invite in this Activity.
+     */
+    private class VoiceBroadcastReceiver extends BroadcastReceiver {
+
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            String action = intent.getAction();
+            if (action.equals(ACTION_OUTGOING_CALL)) {
+                /*
+                 * Handle the incoming call invite
+                 */
+                handleCallRequest(intent);
+            } else if (action.equals(ACTION_DISCONNECT_CALL)) {
+                /*
```

Comment:
These comment blocks are a bit redundant since the action already says what's going to happen